pkg/forklift-api/services/tls_certificate.go

```diff
@@ -0,0 +1,42 @@
+package services
+
+import (
+	"encoding/pem"
+	"net/http"
+
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/container"
+	core "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func serveTlsCertificate(resp http.ResponseWriter, req *http.Request, client client.Client) {
+	if url := req.URL.Query().Get("URL"); url != "" {
```

Comment:
I think if we give up on the if with a statement here we can reduce some of the nesting (not sure if sonarcloud complains)

```go

url := req.URL.Query().Get("URL")
if url == "" {
	msg := "Required parameter is missing: URL"
	if _, err := resp.Write([]byte(msg)); err != nil {
		log.Error(err, "failed to write error message", "message", msg)
	}
	resp.WriteHeader(http.StatusBadRequest)
	return
}
...
```

Also, instead of using `resp.WriteHeader` I believe we can use `http.Error` (we already use it, but not everywhere):
```go
	http.Error(resp, msg, http.StatusBadRequest)
```

Reply:
changed to use http.Error - it's nice that it doesn't return anything, that's why that code got more complicated than it was at the beginning
but it's not complex enough for sonarcloud to complain about :)